site/app/views/UserProfileView.php

```diff
@@ -23,10 +23,10 @@ public function showUserProfile(
         string $csrf_token
     ) {
         $autofill_preferred_name = [$user->getLegalGivenName(), $user->getLegalFamilyName()];
-        if ($user->getPreferredGivenName() != "") {
+        if ($user->getPreferredGivenName() !== "" && $user->getPreferredGivenName() !== null) {
```

Comment:
Can getPreferredGivenName() ever return an empty string given the database constraints?